modules/ROOT/pages/syntax/expressions.adoc

```diff
@@ -1418,6 +1428,69 @@ For verifying that an expression is not of a certain type, the following alterna
 <expr> IS NOT TYPED <TYPE>
 ----
 
+[[type-predicate-expressions-any-and-nothing]]
+=== Use of Any and Nothing types
+
+The `ANY` type is a supertype of all types and will therefore match on everything.
+The `NOTHING` type is however the type containing an empty set of values. This means that it will return `false` for all values.
```

Comment:
```suggestion
On the other hand, the `NOTHING` type is the type containing an empty set of values. This means that it returns `false` for all values.
```

Reply:
A curiosity that may potentially result in an extra example here: is there any good use case for `IS :: NOTHING`? When would I want to check that a variable is of _no_ type? How can I even get a variable that is of no type? :thinking:

Reply:
The use case is for LIST<NOTHING>, so an empty list :)

Reply:
I think the only place you can have something with no type is for empty lists: the inner type of the list is `NOTHING` -> `LIST<NOTHING>`. We don't block the user from writing `IS :: NOTHING` but it'll always be false 🤷

Reply:
> is for empty lists: the inner type of the list is NOTHING -> LIST<NOTHING>. 

And when what I would want to do that? What do I do with a `LIST<NOTHING>`? :thinking: 

> We don't block the user from writing IS :: NOTHING but it'll always be false

Is it a meaningful feature then? :thinking:

Reply:
> What do I do with a `LIST<NOTHING>`?

Confirm if your list is empty or if it has any values in it:
```
[] IS :: LIST<NOTHING> --> true

[null] IS :: LIST<NOTHING> --> false
[1] IS :: LIST<NOTHING> --> false
[true] IS :: LIST<NOTHING> --> false
```

> Is it a meaningful feature then? 🤔

Not to use, but it also feels very odd to block it since it can be an inner type (I guess any examples for it should be with lists but I didn't look that closely on the section to remember what's there), so for consistency it's allowed. (It is also specified as a type in GQL if I remember correctly so it is a general type we should allow the use of.)

Reply:
> Confirm if your list is empty or if it has any values in it:
[] IS :: LIST<NOTHING> --> true

Yeah, although I find much less opaque checking for `var = []`. Shorter and more clearer what its effect is.

Regardless, thanks for clarifying.

Reply:
> that may potentially result in an extra example

I think it might make sense to add an example of `LIST<NOTHING>` showing an empty list as well as a non-empty list to have a 'useful' example case of `NOTHING`

Reply:
This is already there :) Under the list types part (the next section)